apps/pwabuilder/src/script/components/share-card.ts

```diff
@@ -0,0 +1,295 @@
+        }
+
+        .standard-button {
+          margin-bottom: 8px;
+        }
+        
+      `)}
+    `
+  }
+  
+  async setup(){
+    let canvas = (this.shareCanvas.value as HTMLCanvasElement);
+    await draw(canvas, this.manifestData, this.swData, this.securityData, this.siteName);
+    this.dataURL = canvas.toDataURL('image/png', 1.0);
+    this.renderShareActionButtons();
+  }
+
+
+
+  htmlToImage(shareOption: string) {
+
+    if (shareOption === "download"){
+      this.downloadImage(`${this.siteName}_pwabuilder_score.png`)
+    } else if (shareOption === "share"){
+      const file = this.dataURLtoFile(this.dataURL, `${this.siteName}_pwabuilder_score.png`);
+      this.shareFile(file, `${this.siteName} PWABuilder report card score`, "Check out my report card scores from PWABuilder!")
+    } else {  
+      return;
+    }
+    
+  }
+
+  downloadImage(filename: string) {
+    let downloadButtonText = this.shadowRoot!.getElementById("download-button-label");
+    let link = document.createElement('a');
+    link.href = "data:image/png;base64" + this.dataURL;
+    link.download = filename; 
+    link.click();
+    URL.revokeObjectURL(link.href);
+
+    downloadButtonText!.innerText="Downloaded";
+  }
+
+  copyImage() {
+    let canvas = (this.shadowRoot!.getElementById("myCanvas") as HTMLCanvasElement);
+    let copyButtonText = this.shadowRoot!.getElementById("copy-button-label");
+
+    canvas.toBlob(blob => navigator.clipboard
+    .write([new ClipboardItem({'image/png': blob!})])
+    .then(()=>{
+      copyButtonText!.innerText="Copied";
+    }))
+  }
+
+  dataURLtoFile(dataurl: string, filename: string) {
+    var arr = dataurl.split(","),
+    mimeType = arr[0].match(/:(.*?);/)![1],
+    decodedData = atob(arr[1]),
+    lengthOfDecodedData = decodedData.length,
+    u8array = new Uint8Array(lengthOfDecodedData);
+    while (lengthOfDecodedData--) {
+      u8array[lengthOfDecodedData] = decodedData.charCodeAt(lengthOfDecodedData);
+    }
+    return new File([u8array], filename, { type: mimeType });
+  };
+
+  shareFile(file: File, title: string, text: string) {
+    if (navigator.canShare && navigator.canShare({ files: [file] })) {
+      navigator
+        .share({
+          files: [file],
+          title,
+          text
+        })
+        .then(() => console.log("Share was successful."))
+        .catch((error) => console.log("Sharing failed", error));
+    } else {
+      console.log(`Your system doesn't support sharing files.`);
+    }
+  };
+
+  hideDialog(){
+    let dialog: any = this.shadowRoot!.querySelector(".dialog");
+    dialog!.hide();
+  }
+
+  handleRequestClose(e: Event){
+    if (this.preventClosing) {
+      e.preventDefault();
+    }
+  }
+
+  renderShareActionButtons(){
+    const file = this.dataURLtoFile(this.dataURL, `${this.siteName}_pwabuilder_score.png`);
+
+    if (!navigator.canShare || !navigator.canShare({files: [file]})) {
+      this.actionButtons = html`
+        <div>
```

Comment:
This two buttons Copy and Download is completely identical in this IF and in the next ELSE, can you move the condition inside the template and only do on/off for Share button?

Reply:
Added